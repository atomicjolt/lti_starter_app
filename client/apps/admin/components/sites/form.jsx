import React from 'react';
import PropTypes from 'prop-types';
import Input from '../common/input';

export const FIELDS = {
  oauth_key: 'Canvas Developer ID',
  oauth_secret: 'Canvas Developer Key',
  url: 'Canvas Domain',
};

export default function SiteForm(props) {
  const {
    isUpdate,
    oauth_key: oauthKey,
    oauth_secret: oauthSecret,
    url,
    onChange,
    setupSite,
    closeModal,
  } = props;
  const buttonVerb = isUpdate ? 'Update' : 'Create';

  const inputTemplate = (fieldLabel, field, value) => {
    const inputProps = {
      id: `site_${field}`,
      name: field,
      type: 'text',
      value: value || '',
      onChange,
    };
    return (
      <div key={field} className="o-grid__item u-full">
        <Input
          className="c-input"
          labelText={fieldLabel}
          inputProps={inputProps}
        />
      </div>
    );
  };

  return (
    <form>
      <div className="o-grid o-grid__modal-top">
        {inputTemplate(FIELDS.url, 'url', url)}
        {inputTemplate(FIELDS.oauth_key, 'oauth_key', oauthKey)}
        {inputTemplate(FIELDS.oauth_secret, 'oauth_secret', oauthSecret)}
      </div>
      <button
        type="button"
        className="c-btn c-btn--yellow"
        onClick={setupSite}
      >
        {buttonVerb}
        {' '}
        Domain
      </button>
      <button
        type="button"
        className="c-btn c-btn--gray--large u-m-right"
        onClick={closeModal}
      >
        Cancel
      </button>
    </form>
  );
}

SiteForm.propTypes = {
  setupSite: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  isUpdate: PropTypes.bool.isRequired,
  oauth_key: PropTypes.string,
  oauth_secret: PropTypes.string,
  url: PropTypes.string,
};
