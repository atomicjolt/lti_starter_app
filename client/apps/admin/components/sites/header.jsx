import React from 'react';
import PropTypes from 'prop-types';

export default function Header(props) {
  const {
    newSite
  } = props;

  return (
    <div className="c-info">
      <div className="c-title">
        <h1>Sites</h1>
      </div>
      <button className="c-btn c-btn--yellow" onClick={newSite}>
        New Site
      </button>
    </div>
  );
}

Header.propTypes = {
  newSite: PropTypes.func.isRequired,
};
