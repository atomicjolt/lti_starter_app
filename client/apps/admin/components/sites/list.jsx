import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import SiteRow from './site_row';

const List = (props) => {
  const siteRows = _.map(props.sites, (site, index) => (
    <SiteRow
      key={`site_${index}`}
      site={site}
    />
  ));

  if (_.isEmpty(props.sites)) {
    return <p className="c-alert c-alert--danger">No sites are currently configured. Please add a site.</p>;
  }

  return (
    <table className="c-table c-table--lti">
      <thead>
        <tr>
          <th><span>URL</span></th>
          <th><span>SETTINGS</span></th>
          <th><span>DELETE</span></th>
        </tr>
      </thead>
      <tbody>
        {siteRows}
      </tbody>
    </table>
  );
};

List.propTypes = {
  sites: PropTypes.shape({}).isRequired,
};

export default List;
