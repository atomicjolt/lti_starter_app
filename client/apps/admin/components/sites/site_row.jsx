import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import SiteModal from './modal';
import DeleteSiteModal from './delete_modal';

export default class SiteRow extends React.Component {

  static propTypes = {
    site: PropTypes.shape({
      url: PropTypes.string,
      oauth_key: PropTypes.string,
      oauth_secret: PropTypes.string,
    }).isRequired,
  };

  static getStyles() {
    return {
      buttonIcon: {
        border: 'none',
        backgroundColor: 'transparent',
        color: 'grey',
        fontSize: '1.5em',
        cursor: 'pointer',
      },
      alertStyle: {
        fontSize: '10px'
      }
    };
  }

  constructor() {
    super();
    this.state = {
      siteModalOpen: false,
      confirmDeleteModalOpen: false,
    };
  }

  closeSiteModal() {
    this.setState({
      siteModalOpen: false,
    });
  }

  closeDeleteModal() {
    this.setState({
      confirmDeleteModalOpen: false,
    });
  }

  render() {
    const styles = SiteRow.getStyles();
    let warning = null;
    if (_.isEmpty(this.props.site.oauth_key) || _.isEmpty(this.props.site.oauth_secret)) {
      warning = (
        <span className="c-alert c-alert--danger" style={styles.alertStyle}>
          OAuth key and/or secret not configured
        </span>
      );
    }
    return (
      <tr>
        <td>
          {this.props.site.url}
          {warning}
        </td>
        <td>
          <button
            style={styles.buttonIcon}
            onClick={() => this.setState({ siteModalOpen: true })}
          >
            <i className="i-settings" />
          </button>
          <SiteModal
            site={this.props.site}
            isOpen={this.state.siteModalOpen}
            closeModal={() => this.closeSiteModal()}
          />
        </td>
        <td>
          <button
            style={styles.buttonIcon}
            onClick={() => this.setState({ confirmDeleteModalOpen: true })}
          >
            <i className="i-delete" />
          </button>
          <DeleteSiteModal
            site={this.props.site}
            isOpen={this.state.confirmDeleteModalOpen}
            closeModal={() => this.closeDeleteModal()}
          />
        </td>
      </tr>
    );
  }

}
