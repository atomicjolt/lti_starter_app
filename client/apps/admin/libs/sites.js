
// Generates the url where the user can go to setup developer keys for the given site
export function canvasDevKeysUrl(site) {
  return `${site.url}/accounts/site_admin/developer_keys`;
}

export function oauthCallbackUrl(applicationInstanceDomain) {
  return `https://${applicationInstanceDomain}/users/auth/canvas/callback`;
}
