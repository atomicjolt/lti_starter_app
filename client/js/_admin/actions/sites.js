import wrapper from '../../constants/wrapper';
import Network from '../../constants/network';

// Local actions
const actions = [];

// Actions that make an api request
const requests = [
  'GET_SITES',
  'CREATE_SITE',
  'UPDATE_SITE',
  'DELETE_SITE',
];

export const Constants = wrapper(actions, requests);

export function getSites() {
  return {
    type   : Constants.GET_SITES,
    method : Network.GET,
    url    : 'api/sites',
  };
}

export function createSite(site) {
  return {
    type   : Constants.CREATE_SITE,
    method : Network.POST,
    url    : 'api/sites',
    body   : {
      site,
    },
  };
}

export function updateSite(site) {
  return {
    type: Constants.UPDATE_SITE,
    method: Network.PUT,
    url: `api/sites/${site.id}`,
    body: {
      site,
    },
  };
}

export function deleteSite(siteId) {
  return {
    type: Constants.DELETE_SITE,
    method: Network.DEL,
    url: `api/sites/${siteId}`,
    siteId
  };
}
