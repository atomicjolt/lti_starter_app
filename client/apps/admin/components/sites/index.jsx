import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Heading from '../common/heading';
import Header from './header';
import SiteModal from './modal';
import List from './list';
import { deleteSite } from '../../actions/sites';

export default function Index() {

  const [siteModalOpen, setSiteModalOpen] = useState(false);

  const sites = useSelector((state) => state.sites);
  const dispatch = useDispatch();

  const closeSiteModal = () => {
    setSiteModalOpen(false);
  };

  return (
    <div>
      <Heading />
      <div className="o-contain o-contain--full">
        <SiteModal
          isOpen={siteModalOpen}
          closeModal={() => closeSiteModal()}
        />
        <Header
          openSettings={() => {}}
          newSite={() => setSiteModalOpen(true)}
        />
        <List
          sites={sites}
          deleteSite={(siteId) => dispatch(deleteSite(siteId))}
        />
      </div>
    </div>
  );
}

Index.propTypes = {
};
