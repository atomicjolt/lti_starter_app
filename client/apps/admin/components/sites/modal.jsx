import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ReactModal from 'react-modal';

import SiteForm from './form';
import * as SiteActions from '../../actions/sites';
import { canvasDevKeysUrl } from '../../libs/sites';

const select = state => ({
  settings: state.settings,
  siteToOauth: state.siteToOauth
});

export class SiteModal extends React.Component {
  static propTypes = {
    site: PropTypes.shape({
      id: PropTypes.number,
    }),
    createSite: PropTypes.func.isRequired,
    updateSite: PropTypes.func.isRequired,
    isOpen: PropTypes.bool.isRequired,
    closeModal: PropTypes.func.isRequired,
    settings: PropTypes.shape({
      lti_key: PropTypes.string,
      canvas_callback_url: PropTypes.string,
    }).isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      site: props.site || {},
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      site: nextProps.site,
    });
  }

  setupSite() {
    const {
      site,
    } = this.state;
    const isUpdate = !!(site && site.id);

    if (this.hasUrlError()) {
      return;
    }

    if (isUpdate) {
      this.props.updateSite(site);
    } else {
      this.props.createSite(site);
    }
    this.props.closeModal();
  }

  siteChange(e) {
    this.setState({
      site: {
        ...this.state.site,
        [e.target.name]: e.target.value
      }
    });
  }

  hasUrlError() {
    const { site } = this.state;
    return site &&
      site.url &&
      site.url.length > 8 && // Must be longer than just https://
      !_.startsWith(site.url, 'https://');
  }

  render() {
    const { site } = this.state;
    const isUpdate = !!(site && site.id);
    const verb = isUpdate ? 'Update' : 'New';

    const callbackUrl = this.props.settings.canvas_callback_url;
    let canvasDevInstructions = 'Add a canvas domain first.';
    if (site && site.url) {
      canvasDevInstructions = (
        <div>
          <div>
            Use the redirect URI: {callbackUrl}
          </div>
          <div>
            <a href={canvasDevKeysUrl(site)} className="c-modal__subtext-link" target="_blank" rel="noopener noreferrer" >
              Get Canvas Developer Keys Here
            </a>
          </div>
        </div>
      );
    }

    let urlError = null;
    if (this.hasUrlError()) {
      urlError = 'Url must begin with "https://"';
    }

    return (
      <ReactModal
        isOpen={this.props.isOpen}
        onRequestClose={() => this.props.closeModal()}
        contentLabel="Modal"
        overlayClassName="unused"
        className="c-modal c-modal--site is-open"
      >
        <h2 className="c-modal__title">{verb} Domain</h2>
        <h3 className="c-modal__subtext">{canvasDevInstructions}</h3>
        <h3 className="c-modal__subtext">{urlError}</h3>
        <SiteForm
          isUpdate={isUpdate}
          onChange={e => this.siteChange(e)}
          setupSite={() => this.setupSite()}
          closeModal={() => this.props.closeModal()}
          {...this.state.site}
        />
      </ReactModal>
    );
  }
}

export default connect(select, { ...SiteActions })(SiteModal);
