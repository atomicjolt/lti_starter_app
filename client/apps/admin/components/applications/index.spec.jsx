import React from 'react';
import TestUtils from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import Helper from '../../../../specs_support/helper';
import { Index } from './index';
import sites from '../../reducers/sites';

jest.mock('../../libs/assets');
describe('applications index', () => {

  let result;
  let props;
  const sitesData = { 1: { id: 1, oauth_key: 'akey', oauth_secret: 'secret' } };

  beforeEach(() => {
    props = {
      saveApplication: () => {},
      applications: {
        Spiderman: {
          Power1: 'Wall Crawling',
          Power2: 'Spidey Sense'
        }
      }
    };

    result = TestUtils.renderIntoDocument(
      <Provider store={Helper.makeStore({}, { sites: sitesData }, { sites })}>
        <Index {...props} />
      </Provider>
    );
  });

  it('render the application rows', () => {
    const element = TestUtils.findRenderedDOMComponentWithClass(result, 'o-contain o-contain--full');
    expect(element).toBeDefined();
  });

  it('render application', () => {
    const element = TestUtils.findRenderedDOMComponentWithTag(result, 'tbody');
    expect(element.childNodes.length).toBeGreaterThan(0);
  });

});
