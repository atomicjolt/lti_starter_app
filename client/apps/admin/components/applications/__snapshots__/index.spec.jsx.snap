// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications index matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Index
    applications={
      Object {
        "Spiderman": Object {
          "Power1": "Wall Crawling",
          "Power2": "Spidey Sense",
        },
      }
    }
    saveApplication={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Heading) />,
        <div
          className="o-contain o-contain--full"
        >
          <div
            className="c-info"
          >
            <div
              className="c-title"
            >
              <h1>
                LTI Applications
              </h1>
            </div>
          </div>
          <table
            className="c-table c-table--lti"
          >
            <thead>
              <tr>
                <th>
                  <span>
                    NAME
                  </span>
                </th>
                <th>
                  <span>
                    INSTANCES
                  </span>
                </th>
                <th>
                  <span>
                    SETTINGS
                  </span>
                </th>
              </tr>
            </thead>
            <tbody>
              <ApplicationRow
                application={
                  Object {
                    "Power1": "Wall Crawling",
                    "Power2": "Spidey Sense",
                  }
                }
                saveApplication={[Function]}
              />
            </tbody>
          </table>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="c-info"
            >
              <div
                className="c-title"
              >
                <h1>
                  LTI Applications
                </h1>
              </div>
            </div>,
            <table
              className="c-table c-table--lti"
            >
              <thead>
                <tr>
                  <th>
                    <span>
                      NAME
                    </span>
                  </th>
                  <th>
                    <span>
                      INSTANCES
                    </span>
                  </th>
                  <th>
                    <span>
                      SETTINGS
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody>
                <ApplicationRow
                  application={
                    Object {
                      "Power1": "Wall Crawling",
                      "Power2": "Spidey Sense",
                    }
                  }
                  saveApplication={[Function]}
                />
              </tbody>
            </table>,
          ],
          "className": "o-contain o-contain--full",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="c-title"
              >
                <h1>
                  LTI Applications
                </h1>
              </div>,
              "className": "c-info",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                  LTI Applications
                </h1>,
                "className": "c-title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "LTI Applications",
                },
                "ref": null,
                "rendered": "LTI Applications",
                "type": "h1",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <thead>
                  <tr>
                    <th>
                      <span>
                        NAME
                      </span>
                    </th>
                    <th>
                      <span>
                        INSTANCES
                      </span>
                    </th>
                    <th>
                      <span>
                        SETTINGS
                      </span>
                    </th>
                  </tr>
                </thead>,
                <tbody>
                  <ApplicationRow
                    application={
                      Object {
                        "Power1": "Wall Crawling",
                        "Power2": "Spidey Sense",
                      }
                    }
                    saveApplication={[Function]}
                  />
                </tbody>,
              ],
              "className": "c-table c-table--lti",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <tr>
                    <th>
                      <span>
                        NAME
                      </span>
                    </th>
                    <th>
                      <span>
                        INSTANCES
                      </span>
                    </th>
                    <th>
                      <span>
                        SETTINGS
                      </span>
                    </th>
                  </tr>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <th>
                        <span>
                          NAME
                        </span>
                      </th>,
                      <th>
                        <span>
                          INSTANCES
                        </span>
                      </th>,
                      <th>
                        <span>
                          SETTINGS
                        </span>
                      </th>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <span>
                          NAME
                        </span>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "NAME",
                        },
                        "ref": null,
                        "rendered": "NAME",
                        "type": "span",
                      },
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <span>
                          INSTANCES
                        </span>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "INSTANCES",
                        },
                        "ref": null,
                        "rendered": "INSTANCES",
                        "type": "span",
                      },
                      "type": "th",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <span>
                          SETTINGS
                        </span>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "SETTINGS",
                        },
                        "ref": null,
                        "rendered": "SETTINGS",
                        "type": "span",
                      },
                      "type": "th",
                    },
                  ],
                  "type": "tr",
                },
                "type": "thead",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ApplicationRow
                      application={
                        Object {
                          "Power1": "Wall Crawling",
                          "Power2": "Spidey Sense",
                        }
                      }
                      saveApplication={[Function]}
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "Spiderman",
                    "nodeType": "class",
                    "props": Object {
                      "application": Object {
                        "Power1": "Wall Crawling",
                        "Power2": "Spidey Sense",
                      },
                      "saveApplication": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "tbody",
              },
            ],
            "type": "table",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Heading) />,
          <div
            className="o-contain o-contain--full"
          >
            <div
              className="c-info"
            >
              <div
                className="c-title"
              >
                <h1>
                  LTI Applications
                </h1>
              </div>
            </div>
            <table
              className="c-table c-table--lti"
            >
              <thead>
                <tr>
                  <th>
                    <span>
                      NAME
                    </span>
                  </th>
                  <th>
                    <span>
                      INSTANCES
                    </span>
                  </th>
                  <th>
                    <span>
                      SETTINGS
                    </span>
                  </th>
                </tr>
              </thead>
              <tbody>
                <ApplicationRow
                  application={
                    Object {
                      "Power1": "Wall Crawling",
                      "Power2": "Spidey Sense",
                    }
                  }
                  saveApplication={[Function]}
                />
              </tbody>
            </table>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="c-info"
              >
                <div
                  className="c-title"
                >
                  <h1>
                    LTI Applications
                  </h1>
                </div>
              </div>,
              <table
                className="c-table c-table--lti"
              >
                <thead>
                  <tr>
                    <th>
                      <span>
                        NAME
                      </span>
                    </th>
                    <th>
                      <span>
                        INSTANCES
                      </span>
                    </th>
                    <th>
                      <span>
                        SETTINGS
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <ApplicationRow
                    application={
                      Object {
                        "Power1": "Wall Crawling",
                        "Power2": "Spidey Sense",
                      }
                    }
                    saveApplication={[Function]}
                  />
                </tbody>
              </table>,
            ],
            "className": "o-contain o-contain--full",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="c-title"
                >
                  <h1>
                    LTI Applications
                  </h1>
                </div>,
                "className": "c-info",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1>
                    LTI Applications
                  </h1>,
                  "className": "c-title",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "LTI Applications",
                  },
                  "ref": null,
                  "rendered": "LTI Applications",
                  "type": "h1",
                },
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <thead>
                    <tr>
                      <th>
                        <span>
                          NAME
                        </span>
                      </th>
                      <th>
                        <span>
                          INSTANCES
                        </span>
                      </th>
                      <th>
                        <span>
                          SETTINGS
                        </span>
                      </th>
                    </tr>
                  </thead>,
                  <tbody>
                    <ApplicationRow
                      application={
                        Object {
                          "Power1": "Wall Crawling",
                          "Power2": "Spidey Sense",
                        }
                      }
                      saveApplication={[Function]}
                    />
                  </tbody>,
                ],
                "className": "c-table c-table--lti",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <tr>
                      <th>
                        <span>
                          NAME
                        </span>
                      </th>
                      <th>
                        <span>
                          INSTANCES
                        </span>
                      </th>
                      <th>
                        <span>
                          SETTINGS
                        </span>
                      </th>
                    </tr>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <th>
                          <span>
                            NAME
                          </span>
                        </th>,
                        <th>
                          <span>
                            INSTANCES
                          </span>
                        </th>,
                        <th>
                          <span>
                            SETTINGS
                          </span>
                        </th>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <span>
                            NAME
                          </span>,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "NAME",
                          },
                          "ref": null,
                          "rendered": "NAME",
                          "type": "span",
                        },
                        "type": "th",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <span>
                            INSTANCES
                          </span>,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "INSTANCES",
                          },
                          "ref": null,
                          "rendered": "INSTANCES",
                          "type": "span",
                        },
                        "type": "th",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <span>
                            SETTINGS
                          </span>,
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "SETTINGS",
                          },
                          "ref": null,
                          "rendered": "SETTINGS",
                          "type": "span",
                        },
                        "type": "th",
                      },
                    ],
                    "type": "tr",
                  },
                  "type": "thead",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <ApplicationRow
                        application={
                          Object {
                            "Power1": "Wall Crawling",
                            "Power2": "Spidey Sense",
                          }
                        }
                        saveApplication={[Function]}
                      />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": "Spiderman",
                      "nodeType": "class",
                      "props": Object {
                        "application": Object {
                          "Power1": "Wall Crawling",
                          "Power2": "Spidey Sense",
                        },
                        "saveApplication": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "tbody",
                },
              ],
              "type": "table",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
